configuration/azure/appconfig/appconfig.go

```diff
@@ -295,15 +295,21 @@ func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.S
 }
 
 func (r *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, sentinelKey string, id string) {
+	var etagVal *azcore.ETag
 	for {
 		// get sentinel key changes
-		_, err := r.Get(ctx, &configuration.GetRequest{
-			Keys:     []string{sentinelKey},
-			Metadata: req.Metadata,
-		})
+		resp, err := r.client.GetSetting(
+			context.TODO(),
```

Comment:
We have a context already available here, do we require TODO here?

Reply:
Probably should use the context provided here.

Reply:
Actually should be a children context from `ctx` that has a timeout.

`getCtx, getCancel := context.WithTimeout(ctx, timeoutInSeconds)`

Reply:
Thanks for catching. There was one issue that I had created regarding context in az SDK sometime back - https://github.com/Azure/azure-sdk-for-go/issues/19223
[inline comments] https://github.com/dapr/components-contrib/blob/7295b3acc16c6dea35371da39a9df6ef90af16b9/configuration/azure/appconfig/appconfig.go#L206

It was fixed few weeks back, will make the change for this.